vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -416,13 +505,47 @@ void Promotion::ClaimTokens(
 
   promotion->credentials->tokens = json_tokens;
   promotion->credentials->blinded_creds = json_blinded;
-  ledger_->InsertOrUpdatePromotion(
-        promotion->Clone(),
-        [](const ledger::Result _){});
+
+  auto save_callback = std::bind(&Promotion::ClaimTokensSaved,
+      this,
+      _1,
+      braveledger_bind_util::FromPromotionToString(promotion->Clone()),
+      callback);
+
+  ledger_->InsertOrUpdatePromotion(promotion->Clone(), save_callback);
+}
+
+void Promotion::ClaimTokensSaved(
+    const ledger::Result result,
+    const std::string& promotion_string,
+    ledger::ResultCallback callback) {
+  if (result != ledger::Result::LEDGER_OK) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Save failed";
```

Comment:
Same comments as above: can we make message more specific and pass back result if possible?

Reply:
see comment above